src/entities/shelley-validator.js

```diff
@@ -0,0 +1,144 @@
+// @flow
+
+import type { Logger } from 'bunyan'
+import _ from 'lodash'
+import cbor from 'cbor'
+import bs58 from 'bs58'
+import blake from 'blakejs'
+// eslint-disable-next-line camelcase
+import { sha3_256 } from 'js-sha3'
+import { helpers } from 'inversify-vanillajs-helpers'
+
+
+import type { TxType, UtxoInput } from '../blockchain/common'
+import {
+  Database, NetworkConfig, Validator,
+} from '../interfaces'
+import SERVICE_IDENTIFIER from '../constants/identifiers'
+
+
+class ShelleyValidator implements Validator {
+  logger: Logger
+
+  db: Database
+
+  expectedNetworkMagic: number
+
+  constructor(
+    logger: Logger,
+    db: Database,
+    networkConfig: NetworkConfig,
+  ) {
+    this.logger = logger
+    this.db = db
+    this.expectedNetworkMagic = networkConfig.networkMagic()
+  }
+
+  async validateTx(txObj: TxType) {
+    try {
+      this.logger.debug('Shelley tx validator. Ignoring for now: ', JSON.stringify(txObj))
+      // TODO: implement
+      // await this.validateDuplicates(txObj)
```

Comment:
We need to remember that all the functions that would be called here that remain in the file are still very Byron-specific and include Byron CBOR parsing, utxo-only inputs, etc.